pkg/agent/cluster.go

```diff
@@ -54,6 +54,7 @@ type clusterInstallStatusHistory struct {
 	ClusterConsoleRouteCreated                          bool
 	ClusterConsoleRouteURLCreated                       bool
 	ClusterInstallComplete                              bool
+	ValidationHistory                                   map[string]map[string]*validationResultHistory
```

Comment:
I'd be tempted to also define a

```
type ValidationResults map[string]validationResultHistory
```

so that this can be a `map[string]ValidationResults`, and you can also use it when initialising the per-host result maps.